### ISSUE TITLE
NVDA doesn't read text selected with Control+A in Excel, even with UIA enabled

### ISSUE THREAD

**Issue by hdzrvcc0X74** [nvaccess/nvda#18203]
Potentially related issues: [#15843](https://github.com/nvaccess/nvda/issue/15843), [#17411](https://github.com/nvaccess/nvda/issues/17411)

---

NVDA doesn't read text selected with Control+A in Excel, even with UIA enabled

### Steps to reproduce:

Open a blank Excel spreadsheet.
2. Enter any text in cell A1.
3. Press End to move the cursor to the end of the line.
4. Press the Control+A shortcut.

### Actual behavior:

speech viewer enabled
Start  window
Search  window
Search box  edit  blank
e
x
 Excel, App, Press right to view options  1 of 1
enter
Excel
Home  grouping
New  grouping  1 of 3
Templates  list  2 of 3
Blank workbook  Alt,   F  ,   H  ,   Y     2  1 of 9
enter
Book1 - Excel
Sheet Sheet1
selected  A1
f2
selected  A1
edit  blank
cap   T
h
i
s
razmak
i
s
razmak
a
razmak
t
e
s
t
točka
enter
A2
up arrow
This is a test.  overflowing  A1
f2
A1
edit  This is a test.
end
blank
ctrl+a
end
blank
ctrl+a
end
blank
home
cap   T
ctrl+a
This is a test. selected
home
cap   T
This is a test. unselected
ctrl+a
This is a test. selected
alt+shift+tab
speech viewer disabled

### Expected behavior:

NVDA should read the selected text, even if the cursor is positioned at the end of the line.

### NVDA logs, crash dumps and other attachments:

N/A

#### NVDA installed/portable/running from source:

Installed.

#### NVDA version:

2025.1beta8 (2025.1.0.36572)

#### Windows version:

Windows 11 24H2 (AMD64) build 26100.4202

#### Name and version of other software in use when reproducing the issue:

Microsoft Office 16.0.18730.20186

#### Does the issue still occur after restarting your computer?

Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Yes, 2024.4.2. Same behaviors.

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes.

**Comment by Adriani90** [comment]
This occurs only when direct editing in cells is enabled in Excel advanced options, which is the default behavior.
Disabling direct editing in cells in Excel is at least a workaround, though not really satisfactory.
